vent/controller/control_module.py

```diff
@@ -698,45 +637,104 @@ def stop(self):
         if self._save_logs:               # If we kept records, flush the data
             self.dl.close_logfile()
 
+    def interrupt(self):
+        """
+        If a controller seems stuck, this makes a new thread, and starts the main loop.
+        No parameters should have changed.
+        """
+        self._running = threading.Event()           # New thread
+        self._running.clear()
+        self._lock = threading.Lock()
+        self._running.set()
+        self.__thread = threading.Thread(target=self._start_mainloop, daemon=True)
+        try:
+            self.__thread.start()
+        except:
+            pass
+            #TODO RAISE ALERT FOR UI
```

Comment:
@geoffroeder @sneakers-the-rat  This is the interrupt function that restarts the thread. Something like this?

Reply:
i think u should try to clear the existing threading objects before deleting, will propose something in a sec. otherwise seems aight, when would this be called?

Reply:
you are referring to the `clear` and `release`? looks good to me!